hal/mbed_ticker_api.c

```diff
@@ -235,6 +235,12 @@ static void schedule_interrupt(const ticker_data_t *const ticker)
         }
 
         timestamp_t match_tick = compute_tick(ticker, match_time);
+        if (match_tick == queue->tick_last_read) {
```

Comment:
Basically interrupt which is to be scheduled in the past should be caught by the following check:
https://github.com/ARMmbed/mbed-os/blob/ccff46d9a3c960dae72d08c399e214a3eaeb63c7/hal/mbed_ticker_api.c#L232

While debugging issue issue #6054 I did not found case when `match_tick < queue->tick_last_read` which does not mean that such case can not occur. I think that it is worth to consider proposed change (even it it can never happen) since in such case interrupt for sure should be fired immediately.

@c1728p9 what is your opinion?

Reply:
I checked the impact of such change (GCC_ARM/K64F) and `tests-mbed_hal-ticker` tests fails (`test_insert_event_us_outside_overflow_range` test case).

Reply:
Match tick could have rolled over so `<=` cannot be used.

Reply:
I would maybe add a comment line on this, as it's non-obvious - "The time has been checked to be future, but it could still round to the last tick" or something.